pymc/logprob/order.py

```diff
@@ -229,6 +235,72 @@ def find_measurable_max_neg(fgraph: FunctionGraph, node: Node) -> Optional[List[
     return min_rv
 
 
+@node_rewriter(tracks=[Max])
+def find_measurable_max_neg_discrete(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[TensorVariable]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableMaxNeg):
+        return None  # pragma: no cover
+
+    base_var = node.inputs[0]
+
+    if base_var.owner is None:
+        return None
+
+    # Min is the Max of the negation of the same distribution. Hence, op must be Elemwise
+    if not isinstance(base_var.owner.op, Elemwise):
+        return None
+
+    # negation is rv * (-1). Hence the scalar_op must be Mul
+    try:
+        if not (
+            isinstance(base_var.owner.op.scalar_op, Mul)
+            and len(base_var.owner.inputs) == 2
+            and get_underlying_scalar_constant_value(base_var.owner.inputs[0]) == -1
+        ):
+            return None
+    except NotScalarConstantError:
+        return None
```

Comment:
Suggestion: Refactor this into a helper function, since we have this logic in multiple places now